src/commands/report/purls.ts

```diff
@@ -0,0 +1,65 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { Command, Flags, ux } from '@oclif/core';
+
+import { type Sbom, extractPurls } from '../../service/eol/eol.svc.ts';
+import SbomScan from '../scan/sbom.ts';
+
+export default class ReportPurls extends Command {
+  static override description = 'Generate a list of purls from a sbom';
+  static enableJsonFlag = true;
+  static override examples = [
+    '<%= config.bin %> <%= command.id %> --dir=./my-project',
+    '<%= config.bin %> <%= command.id %> --file=path/to/sbom.json',
+    '<%= config.bin %> <%= command.id %> --dir=./my-project --save',
+  ];
+  static override flags = {
+    file: Flags.string({
+      char: 'f',
+      description: 'The file path of an existing cyclonedx sbom to scan for EOL',
+    }),
+    dir: Flags.string({
+      char: 'd',
+      description: 'The directory to scan in order to create a cyclonedx sbom',
+    }),
+    save: Flags.boolean({
+      char: 's',
+      default: false,
+      description: 'Save the list of purls as nes.purls.json',
+    }),
+  };
+
+  public async run(): Promise<string[]> {
+    const { flags } = await this.parse(ReportPurls);
+    const { dir: _dirFlag, file: _fileFlag, save } = flags;
+
+    // Load the SBOM
+    const sbomCommand = new SbomScan(this.argv, this.config);
+    const sbom: Sbom = await sbomCommand.run();
+
+    // Extract purls from SBOM
+    const purls = await extractPurls(sbom);
+
+    ux.action.stop('Scan completed');
+
+    // Print the purls
+    this.log('Found purls:');
+    for (const purl of purls) {
+      this.log(purl);
+    }
+
+    // Save if requested
+    if (save) {
+      const outputPath = path.join(_dirFlag || process.cwd(), 'nes.purls.json');
+      try {
+        fs.writeFileSync(outputPath, JSON.stringify(purls, null, 2));
+        this.log(`\nPurls saved to ${outputPath}`);
+      } catch (error: unknown) {
+        const errorMessage = error instanceof Error ? error.message : 'Unknown error';
```

Comment:
```suggestion
        const errorMessage = error instanceof Error ? error.message : 'Unknown error';
```

I suggest not checking the instance rather checking if the error.message is present. Duck typing > instance checking